init/prepare.sh

```diff
@@ -44,6 +44,12 @@ pushd $SOURCE_DIR/ndt
     export LDFLAGS="-L$BUILD_DIR/build/lib"
     ./bootstrap
     ./configure --enable-fakewww --prefix=$BUILD_DIR/build
+    # Run unit tests on source before making and installing
+    make check
+    if [[ $? -ne 0 ]]; then
```

Comment:
This script sets the bash options: `set -x` and `set -e`. The `-x` option prints every command, and the `-e` option exits immediately if a command exits with a non-zero status. So, in this case, the condition wouldn't be reached.

If we still want to report a specific message, we could use something like:

    make check || (echo "error message" && exit 1)